src/org/infoglue/calendar/actions/CreateResourceAction.java

```diff
@@ -182,9 +163,16 @@ else if(name.equals("mode"))
 		}
         
         return Action.SUCCESS;
-    } 
-    
-    public String getEventIdAsString()
+    }
+
+	private boolean validateFileType(DiskFileItem dfi) {
+		String fileType = dfi.getContentType();
+		log.info("Uploaded file's file type is: " + fileType);
+		List<String> allowedFileType = ResourceController.getController().getFileTypesForAssetKey(getAssetKey());
+		return allowedFileType.contains(fileType);
+	}
```

Comment:
I think the primary purpose of the content type validation is to make sure the editors don't make mistakes.

The mime type should be the mime type of the uploaded file. I haven't heard that mime types can be faked for uploaded files. Could you provide a reference to an article about this? 😄 

Do you know of a better way to check file content types?